low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -357,3 +381,190 @@ def get_nw_status(self, nw_interface, interface):
+        cortx_services = [
+            "motr-free-space-monitor.service",
+            "s3authserver.service",
+            "s3backgroundproducer.service",
+            "s3backgroundconsumer.service",
+            "hare-hax.service",
+            "haproxy.service",
+            "sspl-ll.service",
+            "kibana.service",
+            "csm_agent.service",
+            "csm_web.service",
+            "event_analyzer.service",
+        ]
+        return cortx_services
+
+    def get_external_service_list(self):
+        """Get list of external services for health generation."""
+        # TODO use solution supplied from RE for getting
+        # list of external services.
+        external_services = set(Conf.get(
+            SSPL_CONF, f"{self.SERVICE_HANDLER}>{self.SERVICE_LIST}", []))
+        return external_services
+
+    def get_cortx_service_info(self):
+        """Get cortx service info in required format."""
+        service_info = []
+        cortx_services = self.get_cortx_service_list()
+        for service in cortx_services:
+            response = self.get_systemd_service_info(service)
+            if response is not None:
+                service_info.append(response)
+        return service_info
+
+    def get_external_service_info(self):
+        """Get external service info in required format."""
+        service_info = []
+        external_services = self.get_external_service_list()
+        for service in external_services:
+            response = self.get_systemd_service_info(service)
+            if response is not None:
+                service_info.append(response)
+        return service_info
+
+    @staticmethod
+    def get_service_info_from_rpm(service, prop):
+        """
+        Get specified service property from its corrosponding RPM.
+
+        eg. (kafka.service,'LICENSE') -> 'Apache License, Version 2.0'
+        """
+        systemd_path_list = ["/usr/lib/systemd/system/",
+                             "/etc/systemd/system/"]
+        result = "NA"
+        for path in systemd_path_list:
+            # unit_file_path represents the path where
+            # systemd service file resides
+            # eg. kafka service -> /etc/systemd/system/kafka.service
+            unit_file_path = path + service
+            if os.path.isfile(unit_file_path):
+                # this command will return the full name of RPM
+                # which installs the service at given unit_file_path
+                # eg. /etc/systemd/system/kafka.service -> kafka-2.13_2.7.0-el7.x86_64
+                command = " ".join(["rpm", "-qf", unit_file_path])
+                command = shlex.split(command)
+                service_rpm, _, ret_code = SimpleProcess(command).run()
+                if ret_code != 0:
+                    return result
+                try:
+                    service_rpm = service_rpm.decode("utf-8")
+                except AttributeError:
+                    # TODO add logs
```

Comment:
Logging mechanism is added now.
Start logging the exceptions

Reply:
Logs are added.